components/ws-daemon/pkg/iws/iws.go

```diff
@@ -1104,13 +1105,16 @@ func getCpuResourceInfoV2(mountPoint, cgroupPath string) (*api.Cpu, error) {
 	used := cpuUsage / totalTime * 1000
 
 	quota, period, err := cpu.Max()
-	if err != nil {
+	if errors.Is(err, os.ErrNotExist) {
+		quota = math.MaxUint64
+	} else if err != nil {
 		return nil, err
 	}
 
 	// if no cpu limit has been specified, use the number of cores
 	var limit uint64
 	if quota == math.MaxUint64 {
+		// TODO(toru): we have to check a parent cgroup instead of a host resources
```

Comment:
@utam0k this logic is wrong. What we should do here is to set the lower limit of the class, and if that value is not set, use the `cpuLimits` value from `ws-daemon`

Reply:
@aledbf Thanks for your review. I have a question. this method is used for `gp top`, right? I think this method doesn't relate to the CPU limiting. Please let me know if I am wrong.

Reply:
This method is not involved in cpu limiting, it is only used for the endpoint that gp top/supervisor is calling to get resource information.